app/models/course.rb

```diff
@@ -540,6 +540,7 @@ def is_published?
   def funding_type=(funding_type)
     assign_program_type_service = Courses::AssignProgramTypeService.new
     assign_program_type_service.execute(funding_type, self)
+    save unless errors
```

Comment:
We shouldn't do this here as it's unexpected that setting an attribute will save the object. I think where you put it originally was more along the right lines. In fact, the calling code should use the service itself. This is a weird way to do it altogether 😆